### PR TITLE
NMS-13494: Add OpenAPI docs for AlarmRestService

### PR DESCRIPTION
**Pull request by sogupta-opennms**
### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13494

### CODE REVIEW

**Comment by fooker** [comment]
Please prefix all commits with the issue number followed by a colon (like `NMS-13494: My descriptive commit message`). Same goes for the PR title.

**Comment by fooker** [comment]
Short note: Please make sure, the git author is set correctly. See https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#_your_identity

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    @GET
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, MediaType.APPLICATION_ATOM_XML})
+    @Operation(summary = "Get all types of alaram", description = "Get all types of alaram  ", tags = {"Alarms"})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Alarm not found",
+                    content = @Content)
+    })
+    Response get(@Context final UriInfo uriInfo, @Context final SearchContext searchContext) ;
```
[the remark this answers is not in the document]
Comment (cgorantla) [reply opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java:155]: FIQL query seems to work fine.

**Review by cgorantla — COMMENTED** [review]
Few things noted here.  Since there are too many APIs, I didn't go through each one. 
I will do one more round of verification this week.

Can you fix the `/alarms?comparator=ge&severity=MINOR` issue that I described below.
Let me know if you need any inputs from me.

**Comment by cgorantla** [comment]
> Short note: Please make sure, the git author is set correctly. See https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#_your_identity

@sogupta-opennms I would be good if you can redo the commit with proper username.

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                    content = @Content(schema = @Schema(implementation = Response.class))),
+            @ApiResponse(responseCode = "400", description = "Bad input request",
+                    content = @Content)
+    })
+    Response closeTicket(@Context final SecurityContext securityContext, @PathParam("id") final Integer alarmId) throws Exception;
+
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, MediaType.APPLICATION_ATOM_XML})
+    @Operation(summary = "Get all types of alaram", description = "Get all types of alaram  ", tags = {"Alarms"})
```
[the remark this answers is not in the document]
Comment (sogupta-opennms) [reply opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java]: Done

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Alarm not found",
```
[the remark this answers is not in the document]
Comment (sogupta-opennms) [reply opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java]: Done

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+                    content = @Content)
+    })
+    Response get(@Context final UriInfo uriInfo, @Context final SearchContext searchContext) ;
+
+    @GET
+    @Path("count")
+    @Produces({MediaType.TEXT_PLAIN})
+    @Operation(summary = "Get alarams count", description = "Get a count's all types of alarams", tags = {"Alarms"})
```
[the remark this answers is not in the document]
Comment (sogupta-opennms) [reply opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java]: Done

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Alarm property not found",
+                    content = @Content)
+    })
+    Response getPropertyValues(@PathParam("propertyId") final String propertyId, @QueryParam("q") final String query, @QueryParam("limit") final Integer limit) ;
+
+    @GET
+    @Path("{id}")
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, MediaType.APPLICATION_ATOM_XML})
+    @Operation(summary = "Get id of alaram", description = "Get id of alaram", tags = {"Alarms"})
```
[the remark this answers is not in the document]
Comment (sogupta-opennms) [reply opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java]: Done

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sogupta-opennms) [reply opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java:152]: /alarms?comparator=ge&severity=MINOR  service is giving response

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+    Response get(@Context final UriInfo uriInfo, @Context final SearchContext searchContext) ;
+
+    @GET
+    @Path("count")
+    @Produces({MediaType.TEXT_PLAIN})
+    @Operation(summary = "Get alarams count", description = "Get a count's all types of alarams", tags = {"Alarms"})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Alarm not found",
```
[the remark this answers is not in the document]
Comment (sogupta-opennms) [reply opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/api/AlarmRestApi.java]: Done

**Pull request closed without merging**